pkg/cmd/admin/image/verify-signature.go

```diff
@@ -0,0 +1,368 @@
+			return err
+		} else {
+			o.CurrentUser = me.Name
+		}
+	}
+
+	return nil
+}
+
+// verifySignature verifies the image signature and returns the identity when the signature
+// is valid.
+func (o *VerifyImageSignatureOptions) verifySignature(pc *signature.PolicyContext, img *imageapi.Image, sigBlob []byte) (string, error) {
+	// Pretend that this is the only signature of img, and see what the policy says.
+	memoryImage, err := newUnparsedImage(o.ExpectedIdentity, img, sigBlob)
+	if err != nil {
+		return "", fmt.Errorf("error setting up signature verification: %v", err)
+	}
+	allowed, err := pc.IsRunningImageAllowed(memoryImage)
+	if !allowed && err == nil {
+		return "", errors.New("internal error: signature rejected but no error set")
+	}
+	if err != nil {
+		return "", fmt.Errorf("signature rejected: %v", err)
+	}
+
+	untrustedInfo, err := signature.GetUntrustedSignatureInformationWithoutVerifying(sigBlob)
+	if err != nil {
+		return "", fmt.Errorf("error getting signing key identity: %v", err) // Note that this is also treated as an unverified signature. It really shouldn’t happen anyway.
+	}
+	return untrustedInfo.UntrustedShortKeyIdentifier, nil
+}
+
+// clearSignatureVerificationStatus removes the current image signature from the Image object by
+// erasing all signature fields that were previously set (when image signature was
+// previously verified).
+func (o *VerifyImageSignatureOptions) clearSignatureVerificationStatus(s *imageapi.ImageSignature) {
+	s.Conditions = []imageapi.SignatureCondition{}
+	s.IssuedBy = nil
+}
+
+// fakeDockerTransport is containers/image/docker.Transport, except that it only provides identity information.
+var fakeDockerTransport = dockerTransport{}
+
+type dockerTransport struct{}
+
+func (t dockerTransport) Name() string {
+	return "docker"
+}
+
+// ParseReference converts a string, which should not start with the ImageTransport.Name prefix, into an ImageReference.
+func (t dockerTransport) ParseReference(reference string) (sigtypes.ImageReference, error) {
+	return parseDockerReference(reference)
+}
+
+// ValidatePolicyConfigurationScope checks that scope is a valid name for a signature.PolicyTransportScopes keys
+// (i.e. a valid PolicyConfigurationIdentity() or PolicyConfigurationNamespaces() return value).
+// It is acceptable to allow an invalid value which will never be matched, it can "only" cause user confusion.
+// scope passed to this function will not be "", that value is always allowed.
+func (t dockerTransport) ValidatePolicyConfigurationScope(scope string) error {
+	// FIXME? We could be verifying the various character set and length restrictions
+	// from docker/distribution/reference.regexp.go, but other than that there
+	// are few semantically invalid strings.
+	return nil
+}
+
+// fakeDockerReference is containers/image/docker.Reference, except that only provides identity information.
+type fakeDockerReference struct{ ref reference.Named }
```

Comment:
fake suggests it's used for testing, pick a different name, eg. `identityOnlyDockerReference`, `dummyDockerReference`. I think, I'm leaning towards the former.